src/Collection/Fulfilment/OrderNotesCollection.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Sdk\src\Collection\Fulfilment;
+
+use MyParcelNL\Sdk\src\Concerns\HasApiKey;
+use MyParcelNL\Sdk\src\Concerns\HasUserAgent;
+use MyParcelNL\Sdk\src\Model\Fulfilment\OrderNote;
+use MyParcelNL\Sdk\src\Model\MyParcelRequest;
+use MyParcelNL\Sdk\src\Model\RequestBody;
+use MyParcelNL\Sdk\src\Support\Arr;
+use MyParcelNL\Sdk\src\Support\Collection;
+use Throwable;
+
+class OrderNotesCollection extends Collection
+{
+    use HasUserAgent;
+    use HasApiKey;
+
+    /**
+     * @return self Collection of notes that were saved.
+     */
+    public function save(): self
+    {
+        $notes = [];
+
+        $this->getUniqueOrderUuids()->each(function (string $orderUuid) use (&$notes) {
+            try {
+                $newNotes = $this->saveForOrder($orderUuid);
+            } catch (Throwable $e) {
+                return;
+            }
+
+            $notes = array_reduce($newNotes, static function (array $notes, OrderNote $note) {
+                $notes[] = $note;
+
+                return $notes;
+            }, $notes);
+        });
+
+        return (new self($notes));
+    }
+
+    /**
+     * @return \MyParcelNL\Sdk\src\Collection\Fulfilment\OrderNotesCollection
+     */
+    private function getUniqueOrderUuids(): OrderNotesCollection
+    {
+        return $this->map(function (OrderNote $orderNote) {
+            return $orderNote->getOrderUuid();
+        })->unique();
+    }
+
+    /**
+     * @param  string $orderUuid
+     *
+     * @return array Indexed array holding the saved order notes.
+     * @throws \MyParcelNL\Sdk\src\Exception\AccountNotActiveException
+     * @throws \MyParcelNL\Sdk\src\Exception\ApiException
+     * @throws \MyParcelNL\Sdk\src\Exception\MissingFieldException
+     * @throws \Exception
+     */
+    private function saveForOrder(string $orderUuid): array
+    {
+        $orderNotes = [];
+
+        $this->each(static function (OrderNote $orderNote) use (&$orderNotes, $orderUuid) {
+            if ($orderUuid === $orderNote->getOrderUuid() && $orderNote->validate()) {
```

Comment:
unhappy flow eerst, maar nog liever vervangen door iets als dit:

```php
        $orderNotes = $this
            ->filter(static function (OrderNote $orderNote) use ($orderUuid) {
                return $orderNote->getOrderUuid() === $orderUuid && $orderNote->validate();
            })
            ->map(static function (OrderNote $orderNote) {
                return $orderNote->toApiObject();
            })
            ->values() // dit is zodat de array_values hieronder weg kan
            ->toArray();
```

Het is een paar regels meer, maar wmb wel leesbaarder. reduce kan ook, maar die is toch ook altijd weer lastiger leesbaar